extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -238,17 +237,42 @@ export class AmpAutocomplete extends AMP.BaseElement {
       this.selectHandler_(e);
     });
 
-    let dataPromise = Promise.resolve();
+    let remoteDataPromise = Promise.resolve();
     if (this.element.hasAttribute('src')) {
-      dataPromise = this.getRemoteData_();
+      if (this.sourceData_) {
+        user().warn(TAG, 'Discovered both inline <script> and remote "src"'
+        + ' data. Was providing two datasets intended?');
+      }
+      remoteDataPromise = this.getRemoteData_();
     }
 
-    return dataPromise.then(value => {
-      this.inlineData_ = value || this.inlineData_;
-      this.renderResults_();
+    return remoteDataPromise.then(remoteData => {
+      // If both types of data are provided, display remote data.
+      this.sourceData_ = remoteData || this.sourceData_;
+      this.filterDataAndRenderResults_(this.sourceData_);
     });
   }
 
+  /** @override */
+  mutatedAttributesCallback(mutations) {
+    const src = mutations['src'];
+    if (src === undefined || src === null) {
+      return Promise.resolve();
+    }
+    if (typeof src === 'string') {
+      return this.getRemoteData_().then(remoteData => {
+        this.sourceData_ = remoteData;
+        this.filterDataAndRenderResults_(this.sourceData_, this.userInput_);
+      });
+    }
+    if (typeof src === 'object') {
+      this.sourceData_ = src['items'] || [];
```

Comment:
for both this and the equivalent in `getRemoteData_`, I think we should throw a `user().warn` saying that expected `items` key was not present and therefore an empty result was rendered.